x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -6,43 +6,73 @@
 import React from 'react';
 import { EuiButtonEmpty, EuiToolTip, EuiIcon } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { ErrorCode } from '../../../../../common/anomaly_detection';
+import { useApmPluginContext } from '../../../../hooks/useApmPluginContext';
 import { APIReturnType } from '../../../../services/rest/createCallApmApi';
 import { APMLink } from './APMLink';
 import { getEnvironmentLabel } from '../../../../../common/environment_filter_values';
 import { useUrlParams } from '../../../../hooks/useUrlParams';
 import { useFetcher, FETCH_STATUS } from '../../../../hooks/useFetcher';
+import { useLicense } from '../../../../hooks/useLicense';
 
 export type AnomalyDetectionApiResponse = APIReturnType<
   '/api/apm/settings/anomaly-detection',
   'GET'
 >;
 
-const DEFAULT_DATA = { jobs: [], hasLegacyJobs: false, errorCode: undefined };
+const DEFAULT_DATA = { jobs: [], hasLegacyJobs: false };
 
 export function AnomalyDetectionSetupLink() {
   const { uiFilters } = useUrlParams();
   const environment = uiFilters.environment;
+  const plugin = useApmPluginContext();
+  const canGetJobs = !!plugin.core.application.capabilities.ml.canGetJobs;
```

Comment:
Do we have a guarantee that `plugin.core.application.capabilities.ml` is available if the ML plugin is disabled?

Reply:
This should probabaly be `const canGetJobs = !!plugin.core.application.capabilities.ml?.canGetJobs;`, but i also have a fix out in PR which prevents this component from being rendered when the plugin is not available:
https://github.com/elastic/kibana/pull/73638/files#diff-b70d5e7a299cd90c75ce7997187840b3R87

Reply:
I'll change `ml.` to `ml?.` which, along with @ogupte's change, should take care of it.